### PR TITLE
Fix StdOut in StdIo handling of Unicode.

### LINKED ISSUE #822: StdIo extension mishandles non-ASCII StdOut
The following code is broken using Java 21.0.2 and junit-pioneer 2.2.0:

```java
    @Test
    @StdIo
    public void stdOutTest(StdOut out) {
        String s = "■━━━".repeat(100);
        System.out.println(s);
    }
```

```java
java.lang.IndexOutOfBoundsException: Range [0, 8190) out of bounds for length 2732

	at java.base/jdk.internal.util.Preconditions$3.apply(Preconditions.java:71)
	at java.base/jdk.internal.util.Preconditions$3.apply(Preconditions.java:68)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)
	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)
	at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:684)
	at java.base/java.lang.StringBuffer.append(StringBuffer.java:393)
	at java.base/java.io.StringWriter.write(StringWriter.java:122)
	at org.junitpioneer.jupiter.StdOutputStream.write(StdOutputStream.java:33)
	at java.base/java.io.PrintStream.implWrite(PrintStream.java:643)
	at java.base/java.io.PrintStream.write(PrintStream.java:623)
	at java.base/sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:309)
	at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:381)
	at java.base/sun.nio.cs.StreamEncoder.implWrite(StreamEncoder.java:357)
	at java.base/sun.nio.cs.StreamEncoder.lockedWrite(StreamEncoder.java:158)
	at java.base/sun.nio.cs.StreamEncoder.write(StreamEncoder.java:139)
	at java.base/java.io.OutputStreamWriter.write(OutputStreamWriter.java:219)
	at java.base/java.io.BufferedWriter.implFlushBuffer(BufferedWriter.java:178)
	at java.base/java.io.BufferedWriter.flushBuffer(BufferedWriter.java:163)
	at java.base/java.io.PrintStream.implWriteln(PrintStream.java:848)
	at java.base/java.io.PrintStream.writeln(PrintStream.java:826)
	at java.base/java.io.PrintStream.println(PrintStream.java:1168)
	at stdOutTest(AppTests.java:46)
```

Basically, what I think is happening is that `StdOutputStream` is mishandling the bytes it is receiving:

https://github.com/junit-pioneer/junit-pioneer/blob/98cef28462c8b7ab66231cc5b7e8daef3b329f67/src/main/java/org/junitpioneer/jupiter/StdOutputStream.java#L32-L34

It takes a byte array and its length, then creates a String from it and passes the length unchanged to the StringWriter. However, as this is not ASCII byte array length does not equal String length and this then hits the out-out-bounds exception later.

### PR DESCRIPTION
**Pull request by J08nY**
Fixes #822.

### CODE REVIEW

**Comment by Michael1993** [comment]
I don't think your current test verifies anything - please use assertThatCode(() -> /* your code */).doesNotThrowAnyException();.

File: src/main/java/org/junitpioneer/jupiter/StdOutputStream.java [hunk 1]
```diff
@@ -10,34 +10,34 @@
...
-		writer.write(new String(b, Charset.defaultCharset()), off, len);
+		out.write(b, off, len);
 	}
 
 	/**
 	 * @return the string that was written to {@code System.out} or {@code System.err}
 	 */
 	public String capturedString() {
-		return writer.toString();
+		return out.toString(Charset.defaultCharset());
```
Comment (Michael1993) [inline src/main/java/org/junitpioneer/jupiter/StdOutputStream.java:40]: Is having a `Charset.defaultCharset()` necessary?

File: src/main/java/org/junitpioneer/jupiter/StdOutputStream.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public String capturedString() {
 	 * @return the lines that were written to {@code System.out} or {@code System.err}
 	 */
 	public String[] capturedLines() {
-		var lines = writer.toString().split(StdIoExtension.SEPARATOR, -1);
+		var lines = out.toString(Charset.defaultCharset()).split(StdIoExtension.SEPARATOR, -1);
```
Comment (Michael1993) [inline src/main/java/org/junitpioneer/jupiter/StdOutputStream.java:58]: Same thing here, `Charset.defaultCharset()` might be unnecessary?

File: src/test/java/org/junitpioneer/jupiter/StdIoExtensionTests.java [hunk 3]
```diff
@@ -240,6 +240,14 @@ void catchesNothing(StdIn in, TestReporter reporter) {
 			// ParameterResolutionException: Discovered multiple competing ParameterResolvers
 		}
 
+		@Test
+		@StdIo()
+		@DisplayName("handles Unicode properly")
+		void handlesUnicode(StdOut out) {
+			String s = "■━━━".repeat(100);
+			System.out.println(s);
```
Comment (Michael1993) [inline src/test/java/org/junitpioneer/jupiter/StdIoExtensionTests.java:248]: I'd use 
```suggestion
			assertThatCode(() -> System.out.println(s)).doesNotThrowAnyException();
```

**Comment by Michael1993** [comment]
Thank you for your contribution @J08nY - would you like to add yourself to the README as a contributor?

File: src/main/java/org/junitpioneer/jupiter/StdOutputStream.java (hunk 1 above)
Comment (J08nY) [reply src/main/java/org/junitpioneer/jupiter/StdOutputStream.java:40]: You are right, it is not. The original code had it while constructing the string where it was necessary but here it is not.
```suggestion
		return out.toString();
```

File: src/main/java/org/junitpioneer/jupiter/StdOutputStream.java (hunk 2 above)
Comment (J08nY) [reply src/main/java/org/junitpioneer/jupiter/StdOutputStream.java:58]:
```suggestion
		var lines = out.toString().split(StdIoExtension.SEPARATOR, -1);
```

**Review by Michael1993 — APPROVED** [review]

**Comment by Bukama** [comment]
Thank you for the PR and @Michael1993 for the first round of reviews. I have nothing to add.

**Review by Bukama — APPROVED** [review]

**Pull request merged**